ConnectionManager.class.nut

```diff
@@ -274,28 +289,46 @@ class ConnectionManager {
         }
 
         // Run the global onConnected Handler if it exists
-        if (_onConnect != null) {
-            imp.wakeup(0, function() { _onConnect(); }.bindenv(this));
-        }
+        if (_onConnect) imp.wakeup(0, function() { _onConnect(); }.bindenv(this));
 
         _processQueue();
     }
 
     // Runs whenever we disconnect, or call disconnect()
     function _onDisconnectedFlow(expected) {
+
         // Set the BlinkUp State
         _setBlinkUpState();
 
         // Run the global onDisconnected Handler if it exists
-        if (_onDisconnect != null) {
-            imp.wakeup(0, function() { _onDisconnect(expected); }.bindenv(this));
-        }
+        if (_onDisconnect) imp.wakeup(0, function() { _onDisconnect(expected); }.bindenv(this));
 
         if (_stayConnected) {
             imp.wakeup(0, connect.bindenv(this));
+        } else {
+            // Brutally stop trying to connect
+            server.disconnect();
```

Comment:
Just curious, why is this branch introduced?

If there is a reason to have it, should we also clean _connecting/_connected states here?